docs/source/user-guide/dataset_object.rst

```diff
@@ -3,8 +3,139 @@
 ====================
 The Dataset Object
 ====================
-The dataset is one of the basic blocks of deepchecks. It is a container for the data and its relevant metadata, like special column names (label, date, index, etc).
-Some of the checks allows to use a dataframe directly, but some others requires the metadata in order to run, so they are limited to working only with Datasets.
+
+
+A Quick Introduction
+====================
+
+The Dataset Object is a simple and robust Object to store tabular data and metadata relevant to Machine Learning.
+
+Common information used in Machine Learning that is aggregated in the Dataset:
+ - features
+    List of column names. This is the features that are passed to the model. If not defined, columns not defined as something else is considered a feature.
+ - cat_features
+    List of column names. A subset of the features. Categorical features normally require some preprocessing before being passed to the model.
+ - label
+   Either name of a column, or `pd.Series`. The classes of the classification problem or values of a regression problem.
+ - index_name
+    Name of the index column. This can be useful to track indexes duplicate etc. Like the other arguments, this is not manditory.
+ - date_name
```

Comment:
I think we should specify the accepted format with an example

Reply:
yes, can be good - though this is also something that I think should be explicit and clear in the API Reference (also so we won't need to maintain double...)